### ISSUE TITLE
duplicate symbols error.  if used with grpc-swift at the same time.

### ISSUE THREAD

**Issue by 1amageek** [firebase/firebase-ios-sdk#7508]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.4
  * Firebase SDK version: 7.5.0
  * Installation method: ` Swift Package Manager`
  * Firebase Component: Firestore

### [REQUIRED] Step 2: Describe the problem

When I use grpc-swift and Firestore with SwiftPackageManager, I get the following error.

```
duplicate symbol 'ssl_ctx_st::ssl_ctx_st(ssl_method_st const*)' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_ctx_st::ssl_ctx_st(ssl_method_st const*)' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_ctx_st::~ssl_ctx_st()' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_ctx_st::~ssl_ctx_st()' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_session_st::~ssl_session_st()' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_session_st::~ssl_session_st()' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_st::ssl_st(ssl_ctx_st*)' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_st::ssl_st(ssl_ctx_st*)' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_st::~ssl_st()' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
duplicate symbol 'ssl_st::~ssl_st()' in:
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/CNIOBoringSSL.o
    /Users/1amageek/Library/Developer/Xcode/DerivedData/HOPTestProject-bbsmxhrxezzefnafzdsclfnjqkgd/Build/Products/Debug-iphonesimulator/openssl_grpc.o
ld: 10 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

#### Steps to reproduce:

1. Create new SPM project.
2. Set package.swift
```
import PackageDescription

let package = Package(
    name: "SampleProject",
    platforms: [.iOS(.v10), .macOS(.v10_12), .tvOS(.v10), .watchOS(.v6)],
    products: [
        // Products define the executables and libraries a package produces, and make them visible to other packages.
        .library(
            name: "SampleProject",
            targets: ["SampleProject"]),
    ],
    dependencies: [
        // Dependencies declare other packages that this package depends on.
        // .package(url: /* package url */, from: "1.0.0"),
        .package(name: "grpc-swift", url: "git@github.com:grpc/grpc-swift.git", .upToNextMajor(from: "1.0.0")),
        .package(name: "Firebase", url: "git@github.com:firebase/firebase-ios-sdk.git", .upToNextMajor(from: "7.5.0"))

    ],
    targets: [
        // Targets are the basic building blocks of a package. A target can define a module or a test suite.
        // Targets can depend on other targets in this package, and on products in packages this package depends on.
        .target(
            name: "SampleProject",
            dependencies: [
                .product(name: "GRPC", package: "grpc-swift"),
                .product(name: "FirebaseFirestore", package: "Firebase")
            ]),
        .testTarget(
            name: "SampleProjectTests",
            dependencies: [
                "SampleProject",
                .product(name: "GRPC", package: "grpc-swift"),
                .product(name: "FirebaseFirestore", package: "Firebase")
            ]),
    ]
)

```

#### Relevant Code:

```
import FirebaseFirestore
import GRPC

struct SampleProject {
    var text = "Hello, World!"
}
```

**Comment by paulb777** [comment]
@1amageek Thanks for the detailed report. I'll investigate this week.

**Comment by wilhuff** [comment]
The issue here is that swift-nio-ssl (a component of grpc-swift) has a vendored copy of BoringSSL that conflicts with the one supplied as a dependency of regular gRPC. There's nothing we can really do here, and a better place to discuss this is in grpc/grpc-swift#1134 (which I see you filed after this issue). The fix in apple/swift-nio-ssl#269 seems like it should resolve this, so there's no action to take here.
